### PR TITLE
Get student by email

### PR DESCRIPTION
**Pull request by kathianadiaz**
- get all study programs
- get all universities
- get student by email

### CODE REVIEW

File: src/lib/domain/student/repositories/university_repository.ts [hunk 1]
```diff
@@ -18,6 +18,13 @@ class UniversityRepository implements i.IUniversityRepository {
         return uni
     }
 
+    async getAllUniversity() : Promise<e.IUniversity[]> {
+
+        const uni = db.university.findMany({})
```
Comment (kenneth-rosario) [inline src/lib/domain/student/repositories/university_repository.ts]: You missed an await here

Comment (kathianadiaz) [reply src/lib/domain/student/repositories/university_repository.ts]: thanks, I'll add it now

File: src/lib/domain/student/interfaces.ts [hunk 2]
```diff
@@ -12,12 +12,14 @@ export interface IStudentRepository {
...
 
     getUniversityById: (u_id: number) => Promise<i.IUniversity>
+    getAllUniversity: () => Promise<i.IUniversity[]> 
 
 }
 
 export interface IStudyProgramRepository {
 
     getStudyProgramById: (sp_id: number) => Promise<i.IStudyProgram>
+    getAllStudyProgram: (sp_id: number) => Promise<i.IStudyProgram[]>
```
Comment (kenneth-rosario) [inline src/lib/domain/student/interfaces.ts]: I think you dont need an id

File: src/lib/domain/student/repositories/studyprogram_repository.ts [hunk 3]
```diff
@@ -19,6 +19,13 @@ class StudyProgramRepository implements i.IStudyProgramRepository {
         
     }
     
+    async getAllStudyProgram(sp_id: number) : Promise<e.IStudyProgram[]>{
```
Comment (kenneth-rosario) [inline src/lib/domain/student/repositories/studyprogram_repository.ts]: I think you dont need an id

**Comment by kenneth-rosario** [comment]
Task linked: [CU-cahdhh Get Student By Email Function](https://app.clickup.com/t/cahdhh)

**Review by kenneth-rosario — APPROVED** [review]

**Pull request merged**